Source/Dafny/Resolver.cs

```diff
@@ -16942,11 +16945,12 @@ class MethodCallInformation {
           actuals.Add(b.Actual);
           substMap.Add(formal, b.Actual);
           var what = whatKind + (context is Method ? " in-parameter" : " argument");
-          if (formals.Count != 1) {
-            what += " " + j;
+          var displayName = formal.HasName && !(formal is ImplicitFormal);
+          if (!namesProvidedExplicitly.Contains(formal.Name) && formals.Count() > 1) {
+            what += $" at index {j}";
           }
-          if (formal.HasName && !(formal is ImplicitFormal)) {
-            what += " named '" + formal.Name + "'";
+          if (displayName) {
```

Comment:
There's a little duplication of logic between here and `ResolveCallStmt`, it might be nice to put that in a method.

Reply:
Nice refactoring done.